### PR TITLE
add a comment to native_functions explaining softmax interfaces

### PR DESCRIPTION
**Pull request by nairbv**
Address the review comment made by @gchanan here:

https://github.com/pytorch/pytorch/pull/22456#discussion_r300715866

### CODE REVIEW

File: aten/src/ATen/native/native_functions.yaml [hunk 1]
```diff
@@ -1829,6 +1830,7 @@
 - func: smm(Tensor self, Tensor mat2) -> Tensor
   variants: function, method
 
+# log_softmax doesn't follow kwargs-only convention because BC-breaking when loading jit models that use this operator.
```
Comment (zou3519) [inline aten/src/ATen/native/native_functions.yaml]: s/log_softmax/softmax

File: aten/src/ATen/native/native_functions.yaml [hunk 2]
```diff
@@ -1180,6 +1180,7 @@
     CPU: logspace_cpu_out
     CUDA: logspace_cuda_out
 
+# log_softmax doesn't follow kwargs-only convention because BC-breaking when loading jit models that use this operator.
```
Comment (zou3519) [inline aten/src/ATen/native/native_functions.yaml]: I don't know what "kwargs-only convention" is, but maybe that's just me

File: aten/src/ATen/native/native_functions.yaml (hunk 1 above)
Comment (nairbv) [reply aten/src/ATen/native/native_functions.yaml]: oops, copy/pasted when I changed the wording. fixed.

File: aten/src/ATen/native/native_functions.yaml (hunk 2 above)
Comment (nairbv) [reply aten/src/ATen/native/native_functions.yaml]: re-worded, let me know if that's clearer.

Comment (zou3519) [reply aten/src/ATen/native/native_functions.yaml]: Ah yes. Much clearer now, thanks

File: aten/src/ATen/native/native_functions.yaml [hunk 3]
```diff
@@ -1829,6 +1830,7 @@
 - func: smm(Tensor self, Tensor mat2) -> Tensor
   variants: function, method
 
+# softmax allows positional dtype, unlike most operators, because kwonly is BC-breaking when loading jit models.
```
Comment (zou3519) [inline aten/src/ATen/native/native_functions.yaml]: For my understanding: why is kw-only BC-breaking when loading JIT models? Does the JIT trace positional-kw args as positional args?

Comment (nairbv) [reply aten/src/ATen/native/native_functions.yaml]: Yup. The issue description that caused us to revert (also sent you a link internally):

`F.log_softmax` was tracing to either `torch.log_softmax(logits, dim)` if dtype was not provided, or `torch.log_softmax(logits, dim, dtype)` if dtype wasn't provided.
It's now tracing to either `torch.log_softmax(logits, dim, dtype=annotate(Optional[int], None))` or `torch.log_softmax(logits, dim, dtype=dtype)` respectively.

Comment (zou3519) [reply aten/src/ATen/native/native_functions.yaml]: Cool. Might be good to put that into the comment so that if someone else tries to do the change in the future they won't have to look up this pull request.

**Review by zou3519 — APPROVED** [review]
Could add some more details to the comments but otherwise lgtm

File: aten/src/ATen/native/native_functions.yaml (hunk 3 above)
Comment (nairbv) [reply aten/src/ATen/native/native_functions.yaml]: yeah I kinda want to avoid being excessively verbose though. I think if we add a unit test (https://github.com/pytorch/pytorch/issues/22650), that'll add clearer protection.

Comment (zou3519) [reply aten/src/ATen/native/native_functions.yaml]: Yeah, unit test is great. In general TorchScript should probably have BC unit tests like this
